src/Symfony/Component/RateLimiter/TokenBucket.php

```diff
@@ -70,15 +70,39 @@ public function getExpirationTime(): int
         return $this->rate->calculateTimeForTokens($this->burstSize);
     }
 
-    public function serialize(): string
+    /**
+     * @internal
+     */
+    public function __serialize(): array
     {
-        return serialize([$this->id, $this->tokens, $this->timer, $this->burstSize, (string) $this->rate]);
+        return [$this->id, $this->tokens, $this->timer, $this->burstSize, (string) $this->rate];
     }
 
-    public function unserialize($serialized): void
+    /**
+     * @internal
+     */
+    public function __unserialize(array $data): void
     {
-        [$this->id, $this->tokens, $this->timer, $this->burstSize, $rate] = unserialize($serialized);
+        [$this->id, $this->tokens, $this->timer, $this->burstSize, $rate] = $data;
 
         $this->rate = Rate::fromString($rate);
     }
+
+    /**
+     * @internal
+     */
+    public function __sleep(): array
+    {
+        $this->rate = (string) $this->rate;
```

Comment:
it might be better to use a separate property for this, in case one wants to use the object after __sleep has been called?

Reply:
no need to declare the extra property btw, this is PHP :)